src/openforms/formio/components/custom.py

```diff
@@ -189,6 +189,20 @@ def build_serializer_field(
 class Map(BasePlugin[Component]):
     formatter = MapFormatter
 
+    def mutate_config_dynamically(
+        self, component, submission: Submission, data: DataMapping
+    ) -> None:
+        if component.get("tileLayerIdentifier", False):
+            try:
+                tile_layer = MapTileLayer.objects.get(
+                    identifier=component["tileLayerIdentifier"]
+                )
+            except MapTileLayer.DoesNotExist:
+                return None
+
+            # Add the tile layer url information
+            component["tileLayerUrl"] = tile_layer.url
```

Comment:
With some shortcuts and our beloved :walrus: operator

```suggestion
        if (identifier := component.get("tileLayerIdentifier")) is not None:
            tile_layer = MapTileLayer.objects.filter(identifier=identifier).first()
            if tile_layer is not None:
                # Add the tile layer url information
                component["tileLayerUrl"] = tile_layer.url
```

the fallback to `False` introduces yet another type to handle (now we have `str | None` rather than `str | None | bool`)

Reply:
Ah okay cool. Yeah that `objects.filter` is also pretty nice